src/lib/transition_frontier/persistent_frontier/persistent_frontier.ml

```diff
@@ -152,6 +152,8 @@ module Instance = struct
       External_transition.Validation.wrap transition
       |> External_transition.skip_time_received_validation
            `This_transition_was_not_received_via_gossip
+      |> External_transition.skip_genesis_protocol_state_validation
```

Comment:
I'll add the check. But regarding your point, genesis state hash should be the same no matter how old the persisted frontier is, right? Otherwise the proofs wouldn't verify

Reply:
It would be the same no matter how old the persisted frontier is, but I wanted this to help prevent someone from loading a persistent frontier for one network and a persistent root for another. Pretty rare situation, but I imagine it's something we may hit when we are transferring persisted information between nodes on testnets.